### PR TITLE
BREAKING CHANGE(web-react): Rename Tabs props and clean up its types #DS-1096

### PR DESCRIPTION
**Pull request by crishpeen**

### CODE REVIEW

**Comment by literat** [comment]
@crishpeen Please, rebase this first.

File: packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx [hunk 1]
```diff
@@ -8,21 +8,21 @@ import { withTabsContext } from '../../../../tests/testUtils/withTabsContext';
...
 
-  classNamePrefixProviderTest(() => <TabItem forTab={0} />, 'Tabs__item');
+  classNamePrefixProviderTest(() => <TabItem forTabPane={0} />, 'Tabs__item');
 
-  restPropsTest((props) => <TabItem forTab={0} {...props} />, 'button');
+  restPropsTest((props) => <TabItem forTabPane={0} {...props} />, 'button');
 
   it('should render button tag when there is no href prop', () => {
-    const dom = render(<TabItem forTab={0} />);
+    const dom = render(<TabItem forTabPane={0} />);
```
Comment (curdaj) [inline packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx:18]:
```suggestion
    render(<TabItem forTabPane={0} />);
```

File: packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx [hunk 2]
```diff
@@ -8,21 +8,21 @@ import { withTabsContext } from '../../../../tests/testUtils/withTabsContext';
...
 
     const element = dom.container.querySelector('button') as HTMLElement;
```
Comment (curdaj) [inline packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx:20]:
```suggestion
        const element = screen.getByRole('tab');
```

File: packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx [hunk 3]
```diff
@@ -8,21 +8,21 @@ import { withTabsContext } from '../../../../tests/testUtils/withTabsContext';
...
     expect(element).toHaveClass('Tabs__link');
   });
 
   it('should have ARIA attributtes', () => {
-    const dom = render(<TabItem forTab="test" />);
+    const dom = render(<TabItem forTabPane="test" />);
```
Comment (curdaj) [inline packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx:25]:
```suggestion
    render(<TabItem forTabPane="test" />);
```

File: packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx [hunk 4]
```diff
@@ -8,21 +8,21 @@ import { withTabsContext } from '../../../../tests/testUtils/withTabsContext';
...
 
     const element = dom.container.querySelector('button') as HTMLElement;
```
Comment (curdaj) [inline packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx:27]:
```suggestion
    const element = screen.getByRole('tab');
    
```

File: packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx [hunk 5]
```diff
@@ -32,7 +32,7 @@ describe('TabItem', () => {
 
   it('should handle tab switch when clicked', async () => {
     const selectTabMock = jest.fn();
-    const dom = render(withTabsContext(TabItem, { selectedTabId: 0, selectTab: selectTabMock })({ forTab: 'test' }));
+    const dom = render(withTabsContext(TabItem, { selectedId: 0, selectTab: selectTabMock })({ forTabPane: 'test' }));
```
Comment (curdaj) [inline packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx:35]:
```suggestion
   render(withTabsContext(TabItem, { selectedId: 0, selectTab: selectTabMock })({ forTabPane: 'test' }));
```

File: packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx [hunk 6]
```diff
@@ -32,7 +32,7 @@ describe('TabItem', () => {
...
 
     fireEvent.click(dom.getByRole('tab') as HTMLElement);
```
Comment (curdaj) [inline packages/web-react/src/components/Tabs/__tests__/TabItem.test.tsx:37]:
```suggestion
    fireEvent.click(screen.getByRole('tab'));
```

File: packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx [hunk 7]
```diff
@@ -15,27 +15,25 @@ describe('TabPane', () => {
...
     'TabsPane',
   );
 
   restPropsTest(
-    withTabsContext((props) => <TabPane {...props} tabId="test" />, { selectedTabId: 'test' } as TabsContextType),
+    withTabsContext((props) => <TabPane {...props} id="test" />, { selectedId: 'test' } as TabsContextType),
     'div',
   );
 
   it('should not render tab pane if tab is not selected', () => {
```
Comment (curdaj) [inline packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx:35]:
```
  it('should not render tab pane if tab is not selected', () => {
    render(withTabsContext(TabPane, { selectedId: 'another-tab', selectTab: jest.fn() })({ id: 'test' }));

    expect(screen.queryByRole('tabpanel')).not.toBeInTheDocument();
  });

  it('should render tab pane if tab is selected', () => {
    render(withTabsContext(TabPane, { selectedId: 'test', selectTab: jest.fn() })({ id: 'test' }));
    screen.debug();
    const tabPane = screen.queryByRole('tabpanel');

    expect(tabPane).toBeInTheDocument();
    expect(tabPane).toHaveAttribute('id', 'test');
    expect(tabPane).toHaveAttribute('aria-labelledby', 'test-tab');
    expect(tabPane).toHaveClass('TabsPane');
    expect(tabPane).toHaveClass('is-selected');
  });
```

**Comment by curdaj** [comment]
I've added a few comments in test files, but It could be a topic for discussion about how to write them. So please just take it as an idea.

File: packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx (hunk 7 above)
Comment (literat) [reply packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx:35]: Can you please provide some context for using `queryByRole` or `getByRole`? Is it considered a best practice or is it efficient? I like those suggestions. Just wonder why is better to use it this way. Thanks.

Comment (curdaj) [reply packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx:35]: I would like to discuss this topic with you or possibly make a presentation to unify our approach on how to use it 😉https://testing-library.com/docs/queries/about#priority

Comment (adamkudrna) [reply packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx:35]: > Can you please provide some context for using queryByRole or getByRole? Is it considered a best practice or is it efficient? I like those suggestions. Just wonder why is better to use it this way.

AFAIK:

- `getByRole` fails ➡️ test fails
- `queryByRole` fails ➡️ test is skipped

Comment (adamkudrna) [reply packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx:35]: @curdaj Thank you for keeping the eye on the tests! I think we can test things that should **not** be there more often.

**Review by adamkudrna — COMMENTED** [review]
TBH, I don't quite understand all the changes here, but the fix of the API looks good to me.

I also expect there will be an ongoing discussion about Twig vs. React `Tabs`.

File: packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx (hunk 7 above)
Comment (literat) [reply packages/web-react/src/components/Tabs/__tests__/TabPane.test.tsx:35]: Ok, thanks for enlightening :-) Let's talk about it and use it.

**Review by curdaj — APPROVED** [review]

File: packages/codemods/src/transforms/v2/web-react/tabs-tabitem-tabpane-props.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { API, FileInfo } from 'jscodeshift';
+
+const transform = (fileInfo: FileInfo, api: API) => {
+  const j = api.jscodeshift;
+  const root = j(fileInfo.source);
+
```
Comment (pavelklibani) [inline packages/codemods/src/transforms/v2/web-react/tabs-tabitem-tabpane-props.ts:6]: We are missing check for importing from our library

**Review by pavelklibani — APPROVED** [review]

**Pull request merged**